virtual_rainforest/soil/model.py

```diff
@@ -0,0 +1,108 @@
+"""The `soil.model` module.
+
+The `soil.model` module creates a `SoilModel` class, which extended the base `Model`
+class to be usable to simulate the soil.
+"""
+
+from __future__ import annotations
+
+from typing import Any
+
+from numpy import timedelta64
+
+from virtual_rainforest.core.logger import LOGGER, log_and_raise
+from virtual_rainforest.core.model import BaseModel
+
+
+class InitialisationError(Exception):
+    """Custom exception class for model initialisation failures."""
+
+
+class SoilModel(BaseModel, model_name="soil"):
+    """A class describing the soil model.
+
+    Describes the specific functions and attributes that the soil module should possess.
+    It's very much incomplete at the moment, and only overwrites one function in a
+    pretty basic manner. This is intended as a demonstration of how the class
+    inheritance should be handled for the model classes.
+
+    Args:
+        start_time: Point in time that the model simulation should be started.
+        end_time: Time that the model simulation should end
+        update_interval: Time to wait between updates of the model state.
+        no_layers: The number of soil layers to be modelled.
+
+    Attributes:
+        name: Names the model that is described
+    """
+
+    name = "soil"
+
+    def __init__(self, *args: Any, **kwargs: Any):
+
+        if args[1] < 1:
+            log_and_raise(
+                "There has to be at least one soil layer in the soil model!", ValueError
+            )
+        elif not isinstance(args[1], int):
+            log_and_raise("The number of soil layers must be an integer!", TypeError)
+
+        super(SoilModel, self).__init__(*args)
+        self.no_layers = args[1]
```

Comment:
Follows the comments above:

```suggestion
    def __init__(self, update_interval: timedelta64, no_layers: int, **kwargs: Any):
        no_layers = int(no_layers)
        if no_layers < 1:
            log_and_raise(
                "There has to be at least one soil layer in the soil model!", ValueError
            )

        super(SoilModel, self).__init__(update_interval, **kwargs)
        self.no_layers = no_layers
        self._repr.append("no_layers")
```